### PR TITLE
Add Library Mistake Quest

### PR DESCRIPTION
**Pull request by Emistry**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Quest allow player to waive the transcendent cost.

Quest Walkthrough
https://www.youtube.com/watch?v=vFYlweubU2M

Monster Spawn
http://s010381.blogspot.com/2019/04/blog-post_87.html

<details>

Quest seem to be introduced in KRO around 2012 April I think
img
</details>

**Issues addressed:** 
none

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2532)

**Comment by Emistry** [comment]
I may be wrong with certains area of the quests or maybe behaviors.  
Help me check the translation too, since I am translating based on the TWRO video.
I have less confidence with the OCR scan for the image found. Chinese is easier for me since I can read them. Hopefully it doesn't sound too weird due the direct translate from chinese to english.

Do let me know if any part are wrong.

File: npc/jobs/valkyrie.txt [hunk 1]
```diff
@@ -217,45 +217,165 @@ yuno_in02,88,164,5	script	Metheus Sylphe#Library	2_M_SAGE_B,{
...
 		close;
 	}
 	mes "[Metheus Sylphe]";
 	mes "Welcome to the Library of the Schweicherbil Magic Academy.";
 	mes "Here, we have a countless number of books. Please take your time and look around.";
 	close;
+	
+	OnKill:
+		if (questprogress(5160)) { // Quest #5160 - Library Mistake
+			if (!countitem(6586))
```
Comment (Asheraf) [inline npc/jobs/valkyrie.txt]: This is incorrect, you can get multiple books and they will all get deleted by the npc.

File: npc/jobs/valkyrie.txt [hunk 2]
```diff
@@ -217,45 +217,165 @@ yuno_in02,88,164,5	script	Metheus Sylphe#Library	2_M_SAGE_B,{
...
+									mes "[Metheus Sylphe]";
+									mes "We're still trying to figure out where it escaped to.";
+									mes "I'll inform you when the library guards have any news for it.";
+									break;
+								case 3:
+									mes "[Metheus Sylphe]";
+									mes "Do you understand the problem now?";
+									mes "Or you know other problems?";
+									next;
+									if (select("Sorry.", "Worry about gifts for girlfriend?", "Worry about the lunch?", "Worry about the escaped books?") == 4) {
```
Comment (Asheraf) [inline npc/jobs/valkyrie.txt]: Text need to be added for each of the other cases

File: npc/jobs/valkyrie.txt [hunk 3]
```diff
@@ -217,45 +217,165 @@ yuno_in02,88,164,5	script	Metheus Sylphe#Library	2_M_SAGE_B,{
...
-			mes "Unfortunately, the original copy of the book has been damaged over time. We currently only allow the public to view a copy of the book.";
-			next;
-			mes "[Metheus Sylphe]";
-			mes "Also, in order to preserve the original 'Book of Ymir,' we have decided to accept donations from people who wish to read the copy we have provided.";
-			next;
+			if (RENEWAL) {
+				if (questprogress(5160)) { // Quest #5160 - Library Mistake
+					mes "to confirm the status of the library guards report?";
+					next;
+					switch (select((!countitem(6586)? "Tell me the report.":""), (countitem(6586)? "I found the book.":""), "Donate Zeny")) {
```
Comment (Asheraf) [inline npc/jobs/valkyrie.txt]: Donation case here should be handled on it's own and not fall through to the other one as it has different text.

File: npc/jobs/valkyrie.txt (hunk 1 above)
Comment (Emistry) [reply npc/jobs/valkyrie.txt]: updated

File: npc/jobs/valkyrie.txt (hunk 2 above)
Comment (Emistry) [reply npc/jobs/valkyrie.txt]: updated

File: npc/jobs/valkyrie.txt (hunk 3 above)
Comment (Emistry) [reply npc/jobs/valkyrie.txt]: updated

**Pull request merged**